homeassistant/components/input_boolean/__init__.py

```diff
@@ -153,3 +213,9 @@ def is_on(self):
         """Turn the entity off."""
         self._state = False
         await self.async_update_ha_state()
+
+    async def async_update_config(self, config: typing.Dict) -> None:
+        """Handle when the config is updated."""
+        self._config = config
+        self._state = config.get(CONF_INITIAL, self._state)
```

Comment:
When we update the name of a device, I would not expect it to reset the on/off state.

Reply:
So this one is interesting indeed. It would keep the current state if you update the name, but what should we do if the old configuration did not specify the [`initial`](https://www.home-assistant.io/integrations/input_boolean/#initial) state, but the update does?

Reply:
ignore it.

Reply:
Agreed. We're not resetting the entities any more, so makes more sense to ignore it. Would need to update tests then, as old reload behavior relied on entities being removed, therefore state set accordingly to `initial` parameter.

Reply:
Resolved in https://github.com/home-assistant/home-assistant/pull/30514/commits/fa59681ba59450d78b4cb3d802186a873516804d